pkg/traceql/engine_metrics.go

```diff
@@ -280,8 +287,23 @@ func NewGroupingAggregator(aggName string, innerAgg func() RangeAggregator, by [
 		}
 	}
 
-	return &GroupingAggregator{
-		series:      map[FastValues]RangeAggregator{},
+	switch len(lookups) {
+	case 1:
```

Comment:
Are these off by one to account for the byFunc dynamic one (used by quantile_over_time and histogram_over_time)?  Also the default maxGroupBys for rate/count is 5, which I think will panic here.  I think what we could do is add the FV1 type, increment for byFunc, and choose FV1...5 directly as needed.

Reply:
Yes, that was the reason. I don't know why I didn't think of doing that instead. Changed now